components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -775,12 +781,12 @@ public static NativeObject jsFunction_getSwagger12Resource(Context cx, Scriptabl
      * @param api
      * @param fileHostObject
      * @param create
-     * @return
+     * @return json string of failed Environments
      * @throws APIManagementException
      */
-    private static boolean saveAPI(APIProvider apiProvider, String providerName, API api, 
-    						FileHostObject fileHostObject, boolean create) throws APIManagementException {
-    	boolean success = false;
+    private static String saveAPI(APIProvider apiProvider, String providerName, API api,

```

Comment:
Do we need to seperatly pass the provider name? We can use api object as we have already set provider as api.getId().getProviderName().